### PR TITLE
Two loaders example

### PR DESCRIPTION
**Pull request by leoromanovich**
add example for 2 validation loaders case
add case to tests
bug: found possible bug in train_val_pl.md, tests fail while log_images=True

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -30,6 +30,7 @@ build_readme:
 	cat docs/readme/examples_source/extractor/train.md >> ${README_FILE}
 	cat docs/readme/examples_source/extractor/val.md >> ${README_FILE}
 	cat docs/readme/examples_source/extractor/train_val_pl.md >> ${README_FILE}
+	cat docs/readme/examples_source/extractor/train_2loaders_val.md >> ${README_FILE}
 	cat docs/readme/examples_source/extractor/retrieval_usage.md >> ${README_FILE}
```
Comment (AlekseySh) [inline Makefile:34]: we don't add all the examples to readme, and this particular example is too specific for the main readme

File: tests/test_runs/test_code_from_markdown.py [hunk 2]
```diff
@@ -27,6 +27,11 @@ def find_code_block(file: Path, start_indicator: str, end_indicator: str) -> str
         ("extractor/val.md", "[comment]:vanilla-validation-start\n", "[comment]:vanilla-validation-end\n"),
         ("extractor/train_val_pl.md", "[comment]:lightning-start\n", "[comment]:lightning-end\n"),
         ("extractor/train_val_pl_ddp.md", "[comment]:lightning-ddp-start\n", "[comment]:lightning-ddp-end\n"),
+        (
```
Comment (AlekseySh) [inline tests/test_runs/test_code_from_markdown.py:30]: can you make it one line or it's black? if so, you can turn it off for this particular file

File: docs/readme/examples_source/extractor/train_2loaders_val.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+optimizer = torch.optim.SGD(extractor.parameters(), lr=1e-6)
+criterion = TripletLossWithMiner(margin=0.1, miner=AllTripletsMiner())
+
+
+# 1st validation dataset
+val_dataset_1 = DatasetQueryGallery(df_val, dataset_root=dataset_root,
+                                  transform=get_normalisation_resize_torch(im_size=224))
+val_loader = torch.utils.data.DataLoader(val_dataset_1, batch_size=4)
+metric_callback = MetricValCallback(metric=EmbeddingMetrics(extra_keys=[val_dataset_1.paths_key, ],
```
Comment (AlekseySh) [inline docs/readme/examples_source/extractor/train_2loaders_val.md:33]: you can remove empty arguments for the metric (`[]`), I just did it for debugging

File: docs/readme/examples_source/extractor/train_2loaders_val.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+print(metrics_lightning)
+print(metric_callback.metric.metrics)
+print(metric_callback_2.metric.metrics)
+```
+[comment]:lightning-2loaders-end
+</p>
+</details>
+
+[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1bVUgdBGWvQgCkba2YtaIRVlUQUz7Q60Z?usp=share_link)
```
Comment (AlekseySh) [inline docs/readme/examples_source/extractor/train_2loaders_val.md:58]: I was thinking about stopping creating collab for every example because they become outdated really fast and there is no easy way to test them. So, let's add this example without Colab

**Comment by AlekseySh** [comment]
> bug: found possible bug in train_val_pl.md, tests fail while log_images=True

I think I remember something like that. The bug was reproducible only locally but CI/CD was okay (otherwise the main branch would be red). If you turn logging on again, I can run the tests and we will see what is going on.

Anyway, I will appreciate if you dig into that & understand what is going on even locally

File: tests/test_runs/test_code_from_markdown.py (hunk 2 above)
Comment (leoromanovich) [reply tests/test_runs/test_code_from_markdown.py:30]: Yeah, it's black. Okay, i'll add ignore for this line.

File: docs/readme/examples_source/extractor/train_2loaders_val.md (hunk 4 above)
Comment (leoromanovich) [reply docs/readme/examples_source/extractor/train_2loaders_val.md:58]: That's placeholder colab from previous example =) So yeah, I'll drop it out.
